src/storage/mvcc/reader.rs

```diff
@@ -176,13 +177,22 @@ impl<'a> MvccReader<'a> {
         // Check for locks that signal concurrent writes.
         if let Some(lock) = try!(self.load_lock(key)) {
             if lock.ts <= ts {
-                // There is a pending lock. Client should wait or clean it.
-                return Err(Error::KeyIsLocked {
-                    key: try!(key.raw()),
-                    primary: lock.primary,
-                    ts: lock.ts,
-                    ttl: lock.ttl,
-                });
+                let raw_key = try!(key.raw());
+                let primary_key = lock.primary;
+                if ts == u64::MAX && raw_key == primary_key {
+                    // when ts==MAX(which means it's a point get by pk or unique key),
+                    // and current key is the primary key,
+                    // return latest commit version's value
```

Comment:
It looks strange for MVCC to be mixed with the concept of _pk_ or _unique key_. Maybe better to extract a `get_latest(&mut self, key: &Key)` method.